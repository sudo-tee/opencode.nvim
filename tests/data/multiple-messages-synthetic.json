[
  {
    "type": "custom.emit_events.finished",
    "properties": []
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "id": "msg_001",
        "role": "user",
        "time": {"created": 1000000001},
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Message 1",
        "id": "prt_001_01",
        "messageID": "msg_001",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "id": "msg_002",
        "role": "assistant",
        "time": {"created": 1000000002},
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "id": "msg_003",
        "role": "user",
        "time": {"created": 1000000003},
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Message 3",
        "id": "prt_003_01",
        "messageID": "msg_003",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "id": "msg_004",
        "role": "assistant",
        "time": {"created": 1000000004},
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "id": "msg_005",
        "role": "user",
        "time": {"created": 1000000005},
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Message 5",
        "id": "prt_005_01",
        "messageID": "msg_005",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "id": "msg_006",
        "role": "assistant",
        "time": {"created": 1000000006},
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "custom.emit_events.finished",
    "properties": []
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Response 2 - Part 1",
        "id": "prt_002_01",
        "messageID": "msg_002",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Response 6 - Part 1",
        "id": "prt_006_01",
        "messageID": "msg_006",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Response 4 - Part 1",
        "id": "prt_004_01",
        "messageID": "msg_004",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "custom.emit_events.finished",
    "properties": []
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Response 2 - Part 2",
        "id": "prt_002_02",
        "messageID": "msg_002",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Response 4 - Part 2",
        "id": "prt_004_02",
        "messageID": "msg_004",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Response 6 - Part 2",
        "id": "prt_006_02",
        "messageID": "msg_006",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "custom.emit_events.finished",
    "properties": []
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Response 2 - Part 3",
        "id": "prt_002_03",
        "messageID": "msg_002",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Response 4 - Part 3",
        "id": "prt_004_03",
        "messageID": "msg_004",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Response 6 - Part 3",
        "id": "prt_006_03",
        "messageID": "msg_006",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "custom.emit_events.finished",
    "properties": []
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Response 2 - Part 4 (late arrival)",
        "id": "prt_002_04",
        "messageID": "msg_002",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "custom.emit_events.finished",
    "properties": []
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "text": "Response 4 - Part 4 (late arrival)",
        "id": "prt_004_04",
        "messageID": "msg_004",
        "type": "text",
        "sessionID": "ses_test"
      }
    }
  },
  {
    "type": "custom.emit_events.finished",
    "properties": []
  }
]
