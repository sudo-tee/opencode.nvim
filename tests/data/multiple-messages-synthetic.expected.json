{"lines":["","----","","","Message 1","","----","","","Response 2 - Part 1","","Response 2 - Part 2","","Response 2 - Part 3","","Response 2 - Part 4 (late arrival)","","----","","","Message 3","","----","","","Response 4 - Part 1","","Response 4 - Part 2","","Response 4 - Part 3","","Response 4 - Part 4 (late arrival)","","----","","","Message 5","","----","","","Response 6 - Part 1","","Response 6 - Part 2","","Response 6 - Part 3",""],"actions":[],"extmarks":[[1,2,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":false,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":10,"virt_text":[["▌󰭻 ","OpencodeMessageRoleUser"],[" "],["USER","OpencodeMessageRoleUser"],["","OpencodeHint"],[" (2001-09-09 01:46:41)","OpencodeHint"],[" [msg_001]","OpencodeHint"]],"virt_text_pos":"win_col"}],[2,3,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":true,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col"}],[3,4,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":true,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col"}],[4,7,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":false,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":10,"virt_text":[[" ","OpencodeMessageRoleAssistant"],[" "],["BUILD","OpencodeMessageRoleAssistant"],["","OpencodeHint"],[" (2001-09-09 01:46:42)","OpencodeHint"],[" [msg_002]","OpencodeHint"]],"virt_text_pos":"win_col"}],[5,18,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":false,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":10,"virt_text":[["▌󰭻 ","OpencodeMessageRoleUser"],[" "],["USER","OpencodeMessageRoleUser"],["","OpencodeHint"],[" (2001-09-09 01:46:43)","OpencodeHint"],[" [msg_003]","OpencodeHint"]],"virt_text_pos":"win_col"}],[6,19,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":true,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col"}],[7,20,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":true,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col"}],[8,23,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":false,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":10,"virt_text":[[" ","OpencodeMessageRoleAssistant"],[" "],["BUILD","OpencodeMessageRoleAssistant"],["","OpencodeHint"],[" (2001-09-09 01:46:44)","OpencodeHint"],[" [msg_004]","OpencodeHint"]],"virt_text_pos":"win_col"}],[9,34,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":false,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":10,"virt_text":[["▌󰭻 ","OpencodeMessageRoleUser"],[" "],["USER","OpencodeMessageRoleUser"],["","OpencodeHint"],[" (2001-09-09 01:46:45)","OpencodeHint"],[" [msg_005]","OpencodeHint"]],"virt_text_pos":"win_col"}],[10,35,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":true,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col"}],[11,36,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":true,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col"}],[12,39,0,{"virt_text_hide":false,"virt_text_repeat_linebreak":false,"ns_id":3,"virt_text_win_col":-3,"right_gravity":true,"priority":10,"virt_text":[[" ","OpencodeMessageRoleAssistant"],[" "],["BUILD","OpencodeMessageRoleAssistant"],["","OpencodeHint"],[" (2001-09-09 01:46:46)","OpencodeHint"],[" [msg_006]","OpencodeHint"]],"virt_text_pos":"win_col"}]],"timestamp":1761967800}